### PR TITLE
[SPARKR-160] Support collectAsMap()

### PR DESCRIPTION
**Pull request by hqzizania**
SPARKR-160: Support collectAsMap()

### CODE REVIEW

**Comment by sun-rui** [comment]
@hqzizania the code looks good. Could you add some unit test cases for collectAsMap()?

**Comment by hqzizania** [comment]
@sun-rui Thanks, test cases for collectAsMap() has been added.

File: pkg/R/RDD.R [hunk 1]
```diff
@@ -382,6 +383,29 @@ setMethod("collectPartition",
...
 
+#' @rdname collect-methods
+#' @export
+#' @description
+#' \code{collectAsMap} returns a named list as a map that contains all of the elements
+#' in a key-value pair RDD. 
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' rdd <- parallelize(sc, list(list(1, 2),list(3, 4)), 2L)
```
Comment (concretevitamin) [inline pkg/R/RDD.R]: nit: space after comma

**Comment by hqzizania** [comment]
@concretevitamin  ooops, it has been added.

**Comment by concretevitamin** [comment]
@hqzizania thanks - the doc also needs to be regenerated I guess. Otherwise LGTM.

**Comment by shivaram** [comment]
LGTM - Thanks @hqzizania for the change. Merging this.

**Pull request merged**